Alpaca.Markets/RestClient.cs

```diff
@@ -122,19 +152,46 @@ public void Dispose()
             for(var attempts = 0; attempts < throttler.MaxAttempts; ++attempts)
             {
                 throttler.WaitToProceed();
-
+                HttpStatusCode statusCode = HttpStatusCode.OK;
                 try
                 {
-                    using (var stream = await httpClient.GetStreamAsync(endpointUri))
-                    using (var reader = new JsonTextReader(new StreamReader(stream)))
+                    using (HttpResponseMessage response = await httpClient.GetAsync(endpointUri, HttpCompletionOption.ResponseHeadersRead))
                     {
-                        var serializer = new JsonSerializer();
-                        return serializer.Deserialize<TJson>(reader);
+                        // Adhere to server reported instructions
+                        statusCode = response.StatusCode;
+                        if (statusCode != HttpStatusCode.OK)
```

Comment:
Design issue - in general all this block has no relationship with REST client itself and just relates to throttling logic. I suggest to move this code into RateThrottler class and add method like CheckHttpResponse(HttpResponseMessage) into IThrottler interface.

Reply:
Moved the logic (adjusted as per prior suggestions), into RateThrottler and updated IThrottler and FakeThrottler accordingly.